src/Select/Select.js

```diff
@@ -0,0 +1,539 @@
+  onSelect?: (item: Item, event: SyntheticEvent<>) => void,
+  /** Text displayed when there is no item selected */
+  placeholder?: string,
+  /** Placement of the Select */
+  placement?: 'bottom-end' | 'bottom-start' | 'top-end' | 'top-start',
+  /** Indicates that the user cannot modify the value of the control */
+  readOnly?: boolean,
+  /** Indicates that the user must select a value before submitting a form */
+  required?: boolean,
+  /** The selected item. For use with controlled components. */
+  selectedItem?: Item,
+  /** Available sizes */
+  size?: 'small' | 'medium' | 'large' | 'jumbo',
+  /** Ref for the trigger */
+  triggerRef?: (node: ?React$Component<*, *>) => void,
+  /**
+   * Use a Portal to render the Select menu to the body rather than as a sibling
+   * to the trigger
+   */
+  usePortal?: boolean,
+  /** Available variants */
+  variant?: 'danger' | 'success' | 'warning'
+};
+
+type State = {
+  highlightedIndex: ?number,
+  isOpen?: boolean,
+  selectedItem: ?Item
+};
+
+export const componentTheme = (baseTheme: Object) => ({
+  ...mapComponentThemes(
+    {
+      name: 'Dropdown',
+      theme: dropdownComponentTheme(baseTheme)
+    },
+    {
+      name: 'Select',
+      theme: {}
+    },
+    {
+      ...selectTriggerComponentTheme(baseTheme),
+      ...baseTheme
+    }
+  )
+});
+
+const _Root = createThemedComponent(Dropdown, ({ theme: baseTheme }) => ({
+  ...mapComponentThemes(
+    {
+      name: 'Select',
+      theme: componentTheme(baseTheme)
+    },
+    {
+      name: 'Dropdown',
+      theme: {}
+    },
+    baseTheme
+  )
+}));
+
+const Root = createStyledComponent(
+  _Root,
+  {
+    width: '100%',
+
+    '& > span': {
+      width: '100%'
+    }
+  },
+  {
+    displayName: 'Select'
+  }
+);
+
+const contentWidthModifier = {
+  enabled: true,
+  fn: data => {
+    data.styles.minWidth = pxToEm(224);
+    data.styles.width = pxToEm(data.offsets.reference.width);
+    return data;
+  }
+};
+
+/**
+ * Select is a form control that provides a list of options.
+ */
+export default class Select extends Component<Props, State> {
+  static defaultProps = {
+    placeholder: 'Select...',
+    placement: 'bottom-start',
+    size: 'large'
+  };
+
+  state: State = {
+    highlightedIndex: this.props.defaultHighlightedIndex,
+    isOpen: Boolean(this.props.defaultIsOpen),
+    selectedItem: this.props.defaultSelectedItem
+  };
+
+  id: string = this.props.id || `select-${generateId()}`;
+
+  selectTrigger: ?React$Component<*, *>;
+
+  itemMatcher: any;
+
+  render() {
+    const {
+      data,
+      disabled,
+      modifiers,
+      name,
+      placeholder,
+      readOnly,
+      size,
+      variant,
+      triggerRef,
+      ...restProps
+    } = this.props;
+    const isOpen = this.getControllableValue('isOpen');
+    const selectedItem = this.getControllableValue('selectedItem');
+
+    const rootProps = {
+      id: this.id,
+      ...restProps,
+      data,
+      disabled: disabled || readOnly,
```

Comment:
Why is this correct? Should there be a readOnly prop as well?

Reply:
There is a readOnly prop on Select but not on Dropdown.  It does not make sense for Dropdown.